### PR TITLE
STYLE: Fix or suppress non-iterator-returned errors from pylint

### PR DESCRIPTION
**Pull request by roadswitcher**
- [X] addresses one issue ( `non-iterator-returned` ) from #48855

Found one instance of a test that's deliberately returning an Iterator to fool `is_file_like`, suppressing the warning as a false positive based on nature of test.

### CODE REVIEW

File: pandas/tests/io/test_html.py [hunk 1]
```diff
@@ -1280,6 +1280,8 @@ def seek(self, offset):
             def seekable(self):
                 return True
 
+            # GH 48855 suppressing false positives
+            # pylint: disable-next=non-iterator-returned
```
Comment (mroeschke) [inline pandas/tests/io/test_html.py:1284]: Can this be corrected such that the check and existing test passes?

Comment (roadswitcher) [reply pandas/tests/io/test_html.py:1284]: I'll give that a shot this evening.

Comment (roadswitcher) [reply pandas/tests/io/test_html.py]: After reviewing #17975 ( issue prompting this test), I added `__next__` and changed `__iter__` to remove the warning from pylint while letting existing test pass.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @roadswitcher

**Pull request merged**